contracts/ExtendedRoyaltyUtils.cdc

```diff
@@ -0,0 +1,55 @@
+import FungibleToken from "./utility/FungibleToken.cdc"
+/// This is a version of the royalty views that has been extended with potantial utility functions
+/// It is a work in progress
+pub contract MetadataViews {
+
+    /// Optional interface to provide the helper functions
+    /// for better integration of the royalty standard with the marketplace
+    /// smart contract or provide help to provide better UI/UX.
+
+    //////////////////////////////////////////////////////////////////
+    /// Optional helper functions to facilitate the royalty integration 
+    //////////////////////////////////////////////////////////////////
+
+    /// Allow the sale lister to validate whether the beneficary of the royalty
+    /// is using the compatible capability with what buyer accepts.
+    pub fun checkWalletType(royalties: Royalties, typeToCheck: Type): Bool {
```

Comment:
This will not work with wallet abstractions. Personally I think that when we have default impl in place there should be a new method in FT<Receiver> That checks what Types that wallets accept. It can then be defaulty implemented to return a sane value.